### DISCUSSION TITLE
An efficient way to synchronize?

### DISCUSSION THREAD

**Discussion by CKingX** [abraunegg/onedrive#1921] [Q&A] [answered]
I am trying to synchronize files efficiently. Whenever I ran onedrive --synchronize, it would take a long time to synchronize (especially uploading changes). onedrive --monitor is, of course, much more efficient. So I thought I can run onedrive --monitor as well as onedrive --synchronize --download-only and it works but it can throw another process is locking database. I tried to set up crontab with different delay (5 minutes vs 7 minutes) but that didn't work either. So now I am back at onedrive --synchronize every 10 minutes in crontab. Is there a more effective way (basically monitor that also synchronizes after somehow? I can't use webhook as I will be behind a NAT)

**Comment by abraunegg** [comment, marked as answer]
@CKingX 
> So now I am back at onedrive --synchronize every 10 minutes in crontab. Is there a more effective way (basically monitor that also synchronizes after somehow?

This is *exactly* what running the application in `--monitor` does:

* Monitors your local disk for changes to upload
* Performs a 'sync' with OneDrive every X seconds as governed by the 'monitor_interval' value

In your case - set 'monitor_interval' in your config file to 600 as this equals 10 minutes and configure a systemd service to complete your requirement.

**Reply by CKingX** [reply]
Ok thanks! I got confused thinking it needed a webhook. I will keep it at its default 300 seconds as the only reason I chose 10 minutes was synchronize needed time to compare all files. Monitor does not need that.
